test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,301 @@ def forward(self, x):
             module_entry = per_channel_info["per_channel_status"][key]
             self.assertEqual(module_entry["per_channel_supported"], True)
             self.assertEqual(module_entry["per_channel_used"], False)
+
+
+"""
+Partition on Domain / Things to Test
+
+- All zero tensor
+- Multiple tensor dimensions
+- All of the outward facing functions
+- Epoch min max are correctly updating
+- Batch range is correctly averaging as expected
+- Reset for each epoch is correctly resetting the values
+
+Partition on Output
+- the calcuation of the ratio is occurring correctly
+
+"""
+
+
+class TestModelReportObserver(QuantizationTestCase):
+    class NestedModifiedSingleLayerLinear(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.obs1 = ModelReportObserver()
+            self.mod1 = SingleLayerLinearModel()
+            self.obs2 = ModelReportObserver()
+            self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+            self.relu = torch.nn.ReLU()
+
+        def forward(self, x):
+            x = self.obs1(x)
+            x = self.mod1(x)
+            x = self.obs2(x)
+            x = self.fc1(x)
+            x = self.relu(x)
+            return x
+
+    """ Go through model and reset all model report observers recursively """
+
+    def reset_model_report_observers(self, model):
+        # need to do recursively since it can be nested
+        if isinstance(model, ModelReportObserver):
+            model.reset_batch_and_epoch_values()  # call its reset func
+
+        for child in model.children():
+            self.reset_model_report_observers(child)
```

Comment:
we can use apply I think? https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.apply